src/create-job.ts

```diff
@@ -111,15 +114,24 @@ export type JobDescription = {
  * @param description An object that describes the job to create.
  */
 export default async function createJob(description: JobDescription, pipelineData?: any, encodingProfileData?: any, concurrency: boolean = true) {
-  logger.info(`Creating job ${description.name}.`);
+  logger.info(`Creating job ${description.name}`);
 
-  let pipeline: Pipeline | undefined = undefined;
+  let pipeline: any | undefined = undefined;
   if (pipelineData && encodingProfileData) {
     pipeline = await loadPipelineFromObjects(pipelineData, encodingProfileData);
-  } else if(typeof description.pipeline === 'object') {
+  } else if(instanceOfLocalPipelineConfiguration(description.pipeline)) {
     pipeline = new LocalPipeline({...description.pipeline, ...description.encodingProfile as Record<string,string>} );
-  } else {
+  } else if(typeof description.pipeline === 'string'){
     pipeline = await loadPipeline(description.pipeline, description.encodingProfile as string);
+  } else if(instanceOfEncorePipelineConfiguration(description.pipeline)){
+    pipeline = new EncorePipeline(description.pipeline);
+    const configuration: EncorePipelineConfiguration = description.pipeline;
+    const instance: EncoreInstance = await pipeline.createEncoreInstance(configuration.apiAddress, configuration.token, configuration.instanceId, configuration.profile);
+    await delay(configuration.encoreInstancePostCreationDelay_ms); // Delay required to allow instance to be created before calling it
+    await pipeline.runTranscodeThenAnalyze(instance);
+    logger.info(`Finishing Encore Pipeline Job ${description.name}.`);
+    await pipeline.deleteEncoreInstance(instance, configuration.apiAddress);
+    return
```

Comment:
The Encore specific tasks like setting up a pipeline and transcoding as well as deletion of the instance should be handled by the transcode function. The `runTranscodeThenAnalyze` function should be run when doing the analysis in `analyzeBruteForce` i.e in `analyzeQuality`. Otherwise we are moving away from the core concept of AutoVMAF and the idea behind abstracting the different pipelines in way that it makes it as easy as possible to implement different vendors by just implementing the abstract class `Pipeline`.